e2e/cypress/integration/files_and_attachments/file_previews_spec.js

```diff
@@ -0,0 +1,493 @@
+            });
+
+            // # Close modal
+            cy.get('.modal-close').click();
+        });
+    });
+
+    it('Excel', () => {
+        const properties = {
+            route: 'mm_file_testing/Documents/Excel.xlsx',
+            type: 'excel',
+        };
+        testGenericFile(properties);
+    });
+
+    it('PPT', () => {
+        const properties = {
+            route: 'mm_file_testing/Documents/PPT.pptx',
+            type: 'ppt',
+        };
+        testGenericFile(properties);
+    });
+
+    it('Word', () => {
+        const properties = {
+            route: 'mm_file_testing/Documents/Word.docx',
+            type: 'word',
+        };
+        testGenericFile(properties);
+    });
+
+    it('Text', () => {
+        const route = 'mm_file_testing/Documents/Text.txt';
+        const filename = route.split('/').pop();
+
+        // # Post file in center channel
+        cy.get('#centerChannelFooter').find('#fileUploadInput').attachFile(route);
+        cy.wait(TIMEOUTS.ONE_SEC);
+        cy.get('#create_post').find('.file-preview').within(() => {
+            // * Thumbnail exist
+            cy.get('.post-image__thumbnail > div.text').should('exist');
+        });
+        cy.postMessage('{enter}');
+        cy.wait(TIMEOUTS.ONE_SEC);
+
+        cy.getLastPost().within(() => {
+            cy.get('.post-image__thumbnail').within(() => {
+                // * Image is posted
+                cy.get('.file-icon.text').should('exist').click();
+            });
+        });
+
+        cy.get('.modal-body').within(() => {
+            cy.get('.post-code').get('code').should('exist');
+
+            // # Hover over the image
+            cy.get('.modal-image__content').trigger('mouseover');
+
+            // * Download button should exist
+            cy.findByText('Download').should('exist').parent().then((downloadLink) => {
+                expect(downloadLink.attr('download')).to.equal(filename);
+
+                const fileAttachmentURL = downloadLink.attr('href');
+
+                // * Verify that download link has correct name
+                downloadAttachmentAndVerifyItsProperties(fileAttachmentURL, filename, 'attachment');
+            });
+
+            // # Close modal
+            cy.get('.modal-close').click();
+        });
+    });
+
+    it('MP4', () => {
+        const properties = {
+            route: 'mm_file_testing/Video/MP4.mp4',
+            shouldPreview: true,
+        };
+        testVideoFile(properties);
+    });
+
+    it('AVI', () => {
+        const properties = {
+            route: 'mm_file_testing/Video/AVI.avi',
+            shouldPreview: false,
+        };
+        testVideoFile(properties);
+    });
+
+    it('MKV', () => {
+        const properties = {
+            route: 'mm_file_testing/Video/MKV.mkv',
+            shouldPreview: false,
+        };
+        testVideoFile(properties);
+    });
+
+    it('MOV', () => {
```

Comment:
This fails for me locally. Video element is not found.

Reply:
Can you confirm what browser/OS are you using? In theory, it might be dependent on the browser and OS.

If it is related to the OS, what do you propose we do here? Remove these tests? Adapt them to the environment where the automated tests will be running?

Reply:
@larkox I'm running macOS/chrome locally. The cypress runs on linux/chrome when triggered for nightly run.

I think we should create two test files,
file_previews_spec_1.js -- all passing cases
file_previews_spec_2.js -- all failing cases (so that we can just disable this file if we don't need to get report from it anymore)